### PR TITLE
Complete pycharm plugin to be cross platform

### LINKED ISSUE #4: Add intelij,pycharm,goland,webstorm... plugins
Try to figureout where they are located:
- windows: registry: HKEY_CURRENT_USER\JetBrains\IntelliJ IDEA\<build number>\(Default)
  - To open them, try:
      `pycharm.exe <path>`
- linux: `charm` / ~/pycharm/bin/pycharm.sh 
- mac: /usr/local/bin/charm
or 
/Applications/PyCharm.app/Contents/MacOS/pycharm
or (community):
/Applications/PyCharm\ CE.app/Contents/MacOS/pycharm

### PR DESCRIPTION
**Pull request by agamm**
Added Linux (tested on ubuntu 18 with app store install + manual path)
Added mac (tested from web install dmg)

fixes https://github.com/agamm/comeback/issues/4  
*(only pycharm, but other is just a matter of changing strings for the name of the program (eg 'pycharm' => 'goland'))*

### CODE REVIEW

File: comeback/utils.py [hunk 1]
```diff
@@ -14,14 +14,13 @@ def get_platform():
     return platforms.get(platform.system(), 'other')
 
 
-def run(cmd, wait=False):
+def run(cmd, wait=False, no_shell=False):
```
Comment (yammesicka) [inline comeback/utils.py]: It takes time to understand because of the double negative. Maybe `use_shell`?

File: comeback/plugins/pycharm/main.py [hunk 2]
```diff
@@ -14,39 +14,63 @@ def run_windows(cwd):
...
+        utils.run([pycharm_path, cwd], no_shell=True)
+        return True, 'Found pycharm'
+
+    utils.run(['pycharm-community', cwd], no_shell=True)
+    return True, 'Found pycharm'
+
+
+def run_mac(cwd):
+    apps_path = pathlib.Path('/Applications')
+    apps = list(apps_path.glob("*"))
```
Comment (yammesicka) [inline comeback/plugins/pycharm/main.py]: No need to `list` it right away

File: comeback/plugins/pycharm/main.py [hunk 3]
```diff
@@ -14,39 +14,63 @@ def run_windows(cwd):
...
+    results = list(filter(lambda x: "pycharm" in x.name.lower(), apps))
+    if len(results) == 0:
```
Comment (yammesicka) [inline comeback/plugins/pycharm/main.py]: Use `if not results:`

File: comeback/plugins/pycharm/main.py [hunk 4]
```diff
@@ -14,39 +14,63 @@ def run_windows(cwd):
...
+        return True, 'Found pycharm'
+
+    utils.run(['pycharm-community', cwd], no_shell=True)
+    return True, 'Found pycharm'
+
+
+def run_mac(cwd):
+    apps_path = pathlib.Path('/Applications')
+    apps = list(apps_path.glob("*"))
+    results = list(filter(lambda x: "pycharm" in x.name.lower(), apps))
```
Comment (yammesicka) [inline comeback/plugins/pycharm/main.py]: You can use the `glob` to filter it. I'd use `[Pp]y[Cc]harm`

File: comeback/plugins/pycharm/main.py [hunk 5]
```diff
@@ -14,39 +14,63 @@ def run_windows(cwd):
...
+    return True, 'Found pycharm'
 
 
-def run_plugin(cwd):
+def run_linux(cwd, pycharm_path):
+    if not utils.binary_exists('pycharm-community'):
+        if not pycharm_path:
+            return False, \
+                   'PyCharm not found, please provide pycharm_path option'
+        utils.run([pycharm_path, cwd], no_shell=True)
```
Comment (yammesicka) [inline comeback/plugins/pycharm/main.py]: I think the list is an implementation detail which the plugins should not be bothered with.

Maybe we should `shlex` at `utils.run`?

**Review by yammesicka — CHANGES_REQUESTED** [review]

File: comeback/plugins/pycharm/main.py (hunk 4 above)
Comment (agamm) [reply comeback/plugins/pycharm/main.py]: I tried, but it seemed more simple this way. If you can check it then go ahead and change it :)

Comment (agamm) [reply comeback/plugins/pycharm/main.py]: * simple to understand

File: comeback/plugins/pycharm/main.py (hunk 5 above)
Comment (agamm) [reply comeback/plugins/pycharm/main.py]: You said that you wanted to raise an error, no? I'm the meantime I stuck to the standard.

File: comeback/plugins/pycharm/main.py (hunk 4 above)
Comment (yammesicka) [reply comeback/plugins/pycharm/main.py]: I can't really check it, but 

```
pycharm_dir_pattern = '*[Pp]y[Cc]harm*'
list(apps_path.glob(pycharm_dir_pattern))
```

should absolutely do the trick without performance penalty, and might even make the code cleaner :)

File: comeback/plugins/pycharm/main.py (hunk 5 above)
Comment (yammesicka) [reply comeback/plugins/pycharm/main.py]: Kk

**Review by yammesicka — APPROVED** [review]

**Pull request merged**